internal/pkg/task/config_runner.go

```diff
@@ -104,3 +107,65 @@ func (r *ConfigRunner) validateDependencies() error {
 
 	return nil
 }
+
+// CheckNonZeroExitCode returns the status of the containers part of the given tasks.
+func (r *ConfigRunner) CheckNonZeroExitCode(tasks []*Task) error {
+	var essentialContainers []string
```

Comment:
We could do very similar logic as https://github.com/aws/copilot-cli/pull/3620/files#diff-f4d4181f6b85ffcf797c84472ae4742e03524f9e77606345fe3c77c2629b9b9dR126 !